R/Application.R

```diff
@@ -325,43 +304,30 @@ Application = R6::R6Class(
 
       return(pid)
     },
-    #------------------------------------------------------------------------
     print_endpoints_summary = function() {
       if (length(self$endpoints) == 0) {
         self$logger$warn("", context = "'Application' doesn't have any endpoints")
       }
       self$logger$info("", context = list(endpoints = self$endpoints))
       return(invisible(self))
     },
-    #------------------------------------------------------------------------
-    add_openapi = function(path = "/openapi.yaml", openapi = openapi_create(),
-                           file_path = "openapi.yaml") {
+    add_openapi = function(path = "/openapi.yaml", file_path = "openapi.yaml") {
       checkmate::assert_string(file_path)
       file_path = path.expand(file_path)
 
-      if (!requireNamespace("yaml", quietly = TRUE)) {
-        stop("please install 'yaml' package")
-      }
-
       file_dir = dirname(file_path)
       if (!dir.exists(file_dir)) {
         dir.create(file_dir, recursive = TRUE)
       }
 
-      openapi = c(openapi, list(paths = private$get_openapi_paths()))
-
-      yaml::write_yaml(openapi, file = file_path)
-      # FIXME when http://www.iana.org/assignments/media-types/media-types.xhtml will be updated
+      # when http://www.iana.org/assignments/media-types/media-types.xhtml will be updated
       # for now use  "application/x-yaml":
       # https://www.quora.com/What-is-the-correct-MIME-type-for-YAML-documents
       self$add_static(path = path, file_path = file_path, content_type = "application/x-yaml")
       return(invisible(file_path))
     },
-    #------------------------------------------------------------------------
-    add_swagger_ui = function(path = "/swagger",
-                              path_openapi = "/openapi.yaml",
-                              use_cdn = TRUE,
-                              path_swagger_assets = "/__swagger__/",
+    add_swagger_ui = function(path = "/swagger", path_openapi = "/openapi.yaml",
```

Comment:
Don't we need any check to OpenAPI yaml added before?

Reply:
in theory you can first add swagger-ui resource and then openapi resource

Reply:
I'm too worried about the user.